src/index.ts

```diff
@@ -1,5 +1,125 @@
-export { parseCommandLine } from "./CommandLineParser";
-export { compile, compileFilesWithOptions, transpileString, watchWithOptions } from "./Compiler";
-export { CompilerOptions, LuaLibImportKind, LuaTarget } from "./CompilerOptions";
+import * as fs from "fs";
+import * as path from "path";
+import * as ts from "typescript";
+import { parseConfigFileWithSystem } from "./CommandLineParser";
+import { CompilerOptions } from "./CompilerOptions";
+import { emitTranspiledFiles, OutputFile } from "./Emit";
+import { transpile, TranspiledFile, TranspileResult } from "./Transpile";
+
+export { parseCommandLine, ParsedCommandLine, updateParsedConfigFile } from "./CommandLineParser";
+export * from "./CompilerOptions";
+export * from "./Emit";
+export * from "./LuaAST";
 export { LuaLibFeature } from "./LuaLib";
-export { LuaTranspiler } from "./LuaTranspiler";
+export * from "./LuaPrinter";
+export * from "./LuaTransformer";
+export * from "./Transpile";
+export * from "./TranspileError";
+
+export interface TranspileFilesResult {
+    diagnostics: ts.Diagnostic[];
+    emitResult: OutputFile[];
```

Comment:
I have changed `transpileFiles` and `transpileProject` to use `emitTranspiledFiles`. That's because `emitTranspiledFiles` would do import path resolution later, so I think that would fit high-level API more. Also `emitTranspiledFiles` requires options, which makes using `transpileProject` harder, and later `emitTranspiledFiles`-like thing might require `program` (to do correct root path inference).

Reply:
but we're losing sourcemap and declarations now? Any way we can keep those?

Reply:
They are still there, just as separate files (if options have it enabled)

Reply:
Aha, so emitResult includes `myfile.d.ts` and `myfile.map.lua` too?

Reply:
Yes